### ISSUE TITLE
Height of 0 being set on navbar collapse when menu is closed with toggle

### ISSUE THREAD

**Issue by romans12** [react-bootstrap/react-bootstrap#3394]
There is a 'height: 0px' being set on the navbar collapse when a user closes the mobile menu. If a user has their screen smaller, toggles the menu button, and then expands the screen the navbar will have a height of 0.
<img width="673" alt="screen shot 2018-12-12 at 7 37 11 am" src="https://user-images.githubusercontent.com/6922590/49873237-04839000-fde1-11e8-88f3-512cacdba232.png">

**Comment by martinpsz** [comment]
So is there a solution to the issue? Or is it a work in progress? I can't read the rest of the thread started by Romans12.

**Comment by taion** [comment]
i'm not aware of any work on this ATM.

**Comment by martinpsz** [comment]
Ok, just fyi, I copied and pasted the code from the navbar page to test whether it was an issue on my end. When the navbar.brand is included, the problem does not occur. It only occurs when the navbar.brand component is not present. At least in my case. Using BS CDN v4.2.1.

**Comment by brycejlowe** [comment]
I am seeing this issue as well.  The steps to re-produce this problem are:

1. Use the example code from https://react-bootstrap.github.io/components/navbar/.
1. Remove the Navbar.Brand component.
1. Resize the browser so the collapsed menu is visible.
1. Expand the collapsed menu and then collapse it.
1. Resize the browser so the full menu is visible.

**Expected behavior:**
The menu is rendered and displayed as it was before the browser collapse menu was clicked.

**Actual behavior:**
The div that has the of “navbar-collapse collapse” has an inline style of “height: 0px;”.  The result of this is when the browser is resized back it renders the height based on the contents within it.

**Additional Information:**
In researching this issue, I did observe what @martinpsz mentioned regarding problem not occurring when a Navbar.Brand component was included as a child of the Navbar component.  I do however believe that the underlying issue does occur regardless of whether or not the Navbar.Brand component is included.  I see the inline style being applied in each scenario.  I believe the problem is simply covered up because of the height of the Navbar.Brand component.

I have compared the behavior to vanilla Bootstrap/jQuery and vanilla Bootstrap/jQuery does not exhibit this problem.

I believe I have narrowed down the problem to Collapse.js and the handleExiting function.  The the handleExiting function sets the height to ‘0px’ when I believe it should set it to null.  The actual sizing of the div is set via the CSS classes that are applied.  I have opened a pull request to include the fixes for this and I have modified the automated testing to check for any empty height rather than an explicit height of 0px.
